pkg/orchestrator/provider/aws/scanestimation/pricefetcher.go

```diff
@@ -0,0 +1,331 @@
+	}
+
+	return o.getPricePerUnit(ctx, usageType, "")
+}
+
+func (o *PriceFetcherImpl) GetDataTransferCostPerGB(sourceRegion, destRegion string) (float64, error) {
+	if destRegion == sourceRegion {
+		return 0, nil
+	}
+	// TODO (erezf) currently I could not find a reliable way to get the data transfer cost from the offer file.
+	// 0.02 seems to be a unified price across regions according to https://aws.amazon.com/ec2/pricing/on-demand/
+	// This price will probably change in the future, so need to think of a way to keep it updated.
+	return 0.02, nil
+}
+
+func (o *PriceFetcherImpl) GetInstancePerHourCost(ctx context.Context, regionCode string, instanceType ec2types.InstanceType, marketOption MarketOption) (float64, error) {
+	if marketOption == MarketOptionSpot {
+		return o.getSpotInstancePerHourCost(ctx, regionCode, instanceType)
+	}
+
+	var usageType string
+	if regionCode == usEast1RegionCode {
+		usageType = fmt.Sprintf("BoxUsage:%v", instanceType)
+	} else {
+		abb, ok := regionCodeToUsageTypeAbbreviation[regionCode]
+		if !ok {
+			return 0, fmt.Errorf("failed to find usage type in regionCodeToUsageTypeAbbreviation map with regionCode %v", regionCode)
+		}
+		// example: USE2-BoxUsage:t2.large
+		usageType = fmt.Sprintf("%v-BoxUsage:%v", abb, instanceType)
+	}
+
+	return o.getPricePerUnit(ctx, usageType, "RunInstances")
+}
+
+func (o *PriceFetcherImpl) getSpotInstancePerHourCost(ctx context.Context, regionCode string, instanceType ec2types.InstanceType) (float64, error) {
+	timeNow := time.Now()
+	ret, err := o.ec2Client.DescribeSpotPriceHistory(ctx, &ec2.DescribeSpotPriceHistoryInput{
+		Filters: []ec2types.Filter{
+			{
+				Name:   utils.PointerTo("product-description"),
+				Values: []string{"Linux/UNIX"},
+			},
+		},
+		InstanceTypes: []ec2types.InstanceType{instanceType},
+		StartTime:     utils.PointerTo(timeNow.Add(-12 * time.Hour)), // last 12 hours should be enough as we only want the latest.
+	}, func(options *ec2.Options) {
+		options.Region = regionCode
+	})
+	if err != nil {
+		return 0, fmt.Errorf("failed to describe spot price history: %v", err)
+	}
+
+	if len(ret.SpotPriceHistory) == 0 {
+		return 0, fmt.Errorf("failed to find spot instances history")
+	}
+
+	// take the latest price (most updated)
+	spotPrice := ret.SpotPriceHistory[0].SpotPrice
+
+	price, err := strconv.ParseFloat(*spotPrice, 64)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse float %v: %v", *spotPrice, err)
+	}
+
+	return price, nil
+}
+
+func (o *PriceFetcherImpl) getPricePerUnit(ctx context.Context, usageType, operation string) (float64, error) {
+	filters := createGetProductsFilters(usageType, "AmazonEC2", operation)
+
+	products, err := o.pricingClient.GetProducts(ctx, &pricing.GetProductsInput{
+		ServiceCode:   utils.PointerTo("AmazonEC2"),
+		Filters:       filters,
+		FormatVersion: utils.PointerTo("aws_v1"),
+	}, func(options *pricing.Options) {
+		options.Region = usEast1RegionCode
+	})
```

Comment:
Do you know if this uses the bulk product API or the product query API. https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/using-pelong.html it looks like it might be the query API which has regional variants for the US, Europe and Asia.

Reply:
Looks like the aws-sdk-v2 is using the query APIs not the bulk ones: https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/pricing#pkg-overview

Only the US and Asia endpoints are supported though:
> You can use the following endpoints for the Amazon Web Services Price List API:
> https://api.pricing.us-east-1.amazonaws.com/
> https://api.pricing.ap-south-1.amazonaws.com/

Lets add a comment that just highlight that the Pricing API is only available on us-east-1 and ap-south-1, and for now we've chosen to use the us-east-1 endpoint.